crates/ruff/src/message/diff.rs

```diff
@@ -0,0 +1,194 @@
+use crate::message::Message;
+use colored::{Color, ColoredString, Colorize, Styles};
+use ruff_diagnostics::Fix;
+use ruff_python_ast::source_code::{OneIndexed, SourceCode};
+use ruff_python_ast::types::Range;
+use ruff_text_size::{TextRange, TextSize};
+use similar::{ChangeTag, TextDiff};
+use std::fmt::{Display, Formatter};
+use std::num::NonZeroUsize;
+
+/// Renders a diff that shows the code fixes.
+///
+/// The implementation isn't fully fledged out and only used by tests. Before using in production, try
```

Comment:
What do we do in production? Continue to show just the changed line?

Reply:
Oh right, sorry, in production we don't show suggested fixes at all, we only show the suggestion _message_, and highlight the relevant code.

Reply:
I do wonder if there's anything we can leverage from Clippy or rustc to facilitate printing diffs.

Reply:
We can look into it when redoing Diagnostics. I would like to replace annotate snippet and colored. Ideally, the code frame and diff is something we build ourselves as it is an important core component of all our tools. This is something that we could inherit from Rome